Kernel/Bus/USB/USBConfiguration.cpp

```diff
@@ -50,11 +54,39 @@ ErrorOr<void> USBConfiguration::enumerate_interfaces()
             // FIXME: It looks like HID descriptors come BEFORE the endpoint descriptors for a HID device, so we should load
             // these too eventually.
             // See here: https://www.usb.org/defined-class-codes
-            if (interface_descriptor->interface_class_code == USB_CLASS_HID)
+            if (interface_descriptor->interface_class_code == USB_CLASS_HID) {
+                USBEndpointDescriptor endpoint_descriptor;
+                AK::TypedTransfer<u8>::copy(reinterpret_cast<u8*>(&endpoint_descriptor), raw_endpoint_descriptor_offset, sizeof(USBEndpointDescriptor));
+
+                if constexpr (USB_DEBUG) {
+                    dbgln("Endpoint Descriptor {}", endpoint);
+                    dbgln("Endpoint Address: {}", endpoint_descriptor.endpoint_address);
+                    dbgln("Endpoint Attribute Bitmap: {:08b}", endpoint_descriptor.endpoint_attributes_bitmap);
+                    dbgln("Endpoint Maximum Packet Size: {}", endpoint_descriptor.max_packet_size);
+                    dbgln("Endpoint Poll Interval (in frames): {}", endpoint_descriptor.poll_interval_in_frames);
+                }
+
+                // FIXME: Allow to set different configurations...
+                USBConfigurationDescriptor descriptor;
+                AK::TypedTransfer<USBConfigurationDescriptor>::copy(&descriptor, &m_descriptor, 1);
+                transfer_length = TRY(m_device.control_transfer(USB_REQUEST_TRANSFER_DIRECTION_DEVICE_TO_HOST, USB_REQUEST_SET_CONFIGURATION, 1, 0, sizeof(USBConfigurationDescriptor), &descriptor));
+                dbgln_if(USB_DEBUG, "USB device configuration was set!");
+
+                // FIXME: Detect USB HID mouse in an abstracted way...
+                if (interface_descriptor->interface_sub_class_code == 0x1 && interface_descriptor->interface_protocol == 0x2) {
```

Comment:
This code shouldn't be in here, but in `USBPipe` instead. See the USB MassStorage PR

Reply:
I don't see how @Hendiadyoin1 code did anything to change `USB::Pipe` to actually detect USB devices?

Reply:
IIRC the only thing I did to Pipes was make the interface nicer